test/books.test.js

```diff
@@ -1,71 +1,156 @@
 const cds = require('@sap/cds');
 
+
+const test = cds.test(__dirname + '/..');
+const { POST, GET, PUT, DELETE, expect, axios } = test;
+const newBook = {
+  title: 'kozina',
+  author: 'saaid'
+};
+const newPurchaseHistory = {
+  book_id: "7812e4c4-9db5-5687-bb64-1b216bb2f569",
+  user_id: "7812e4c4-874d-2365-opl9-09876bb2f790",
+  date: "2022-12-31"
+};
+
+
+let url = (entity, id) => {
+  return id ? `/odata/v4/catalog/${entity}('${id}')` : `/odata/v4/catalog/${entity}`;
+}
+let bookTestId;
+let purchaseHistorTestId;
+
+
+axios.defaults.auth = { username: "fahd", password: "123456" };
+
+const tryCatchBloc = async (fn, err_stratus, message) => {
```

Comment:
this is great to reuse ! 👍  But rename the function with more explicit detail like `executeRequestExpectingErrorStatus`
In addition add a comment explaining the function expects the fn to throw an error with a specific status otherwise the test will fail.